src/index.js

```diff
@@ -19,664 +20,520 @@ const debug = (function () {
-    const currentUserInfo = window.sessionStorage.getItem(CURRENT_USER_INFO_STR)
-    if (!currentUserInfo) return null
-    return JSON.parse(currentUserInfo).profileImage
+  async del (key) {
+    const hashedKey = await common.utils.hashKey(
+      key,
+      this.service.machine.context.userAppNonce
+    )
+    return this.service.machine.context.userAppDb.delAsync(hashedKey)
   }
 
-  async _genLoginLink () {
-    this.loginChannel = uuidv4()
-    this.loginKey = await common.crypto.genAESKey(true, 'AES-GCM', 128)
-    const extractedKey = await common.crypto.exportKey(this.loginKey)
-    const keyBase64 = Buffer.from(extractedKey).toString('base64')
-    const loginUrl = new URL(this.config.masqAppBaseUrl)
-    const requestType = 'login'
-    const hashParams = JSON.stringify([this.appName, requestType, this.loginChannel, keyBase64])
-    loginUrl.hash = '/link/' + Buffer.from(hashParams).toString('base64')
-
-    this.loginLink = loginUrl.href
+  async list (prefix) {
+    return common.utils.list(
+      this.service.machine.context.userAppDb,
+      this.service.machine.context.importedUserAppDEK,
+      this.service.machine.context.userAppNonce,
+      prefix
+    )
   }
 
-  async getLoginLink () {
-    return this.loginLink
+  async watch (key, cb) {
+    return common.utils.watch(
+      this.service.machine.context.userAppDb,
+      this.service.machine.context.userAppNonce,
+      key,
+      cb
+    )
   }
+}
 
-  async logIntoMasq (stayConnected) {
-    await this.setState('loggingIn')
-
-    // make stayConnected i boolean
-    this.stayConnected = !!stayConnected
-
-    let pr = new Promise((resolve, reject) => {
-      // Subscribe to channel for a limited time to sync with masq
-      debug(`Creation of a hub with ${this.loginChannel} channel name`)
-
-      const hub = signalhub(this.loginChannel, this.config.hubUrls)
-      hub.on('error', ({ url, error }) => {
-        reject(new MasqError(MasqError.SIGNALLING_SERVER_ERROR, url, error))
-      })
-
-      this.loginSw = this._createSwarm(hub)
-
-      this.loginSw.on('peer', (peer, id) => {
-        debug(`The peer ${id} join us...`)
-        this.loginPeer = peer
-
-        this.loginPeer.once('data', async (data) => {
-          try {
-            // decrypt the received message and check if the right key has been used
-            const json = await common.crypto.decrypt(this.loginKey, JSON.parse(data), 'base64')
-
-            switch (json.msg) {
-              case 'authorized':
-                await this._receivedAuthorized(json)
-                break
+const _resetLogin = () => {
+  console.log('--- RESET LOGIN ---')
+}
 
-              case 'notAuthorized':
-                // if this User-app is not registered
-                await this._receiveNotAuthorized()
-                break
+const _init = async (context, event, actionMeta) => {
+  // override config with options argument
+  context.config = context.config ? context.config : {}
```

Comment:
this is extra if you affect a default value in constructor `config = {}`

Reply:
we should not modify the object directly (functional obviously)
Instead, we can just write: 
```js
assign({
    hubUrls: context.config.hubUrls ||  jsonConfig.hubUrls,
    masqAppBaseUrl: context.config.masqAppBaseUrl || jsonConfig.masqAppBaseUrl,
    swarmConfig: context.config.swarmConfig || jsonConfig.swarmConfig,
    ...initNullAttributes(context), // this should just return an object, can be changed to getNullAttributes()
})
```